ARCHITECTURE.md

```diff
@@ -15,21 +15,23 @@ our [API documentation][hoomd_documentation] on readthedocs.
 
 ### Continuous integration
 
-[Azure Pipelines][azp_docs] performs continuous integration testing on
-HOOMD-blue. Azure Pipelines compiles HOOMD-blue, runs the unit, validation, and
-style tests and reports the status to GitHub pull requests. A number of parallel
-builds test a variety of compiler and build configurations, including:
+[Github Actions] performs continuous integration testing on HOOMD-blue. GitHub
+Actions compiles HOOMD-blue, runs the unit and validation, and and reports the
+status to GitHub pull requests. A number of parallel builds test a variety of
+compiler and build configurations, including:
 
 * The 2 most recent **CUDA** toolkit versions
 * **gcc** and **clang** versions including the most recent releases back to the
   defaults provided by the oldest maintained Ubuntu LTS release.
 
-Visit the [glotzerlab/hoomd-blue][hoomd_builds] pipelines page to find recent
-builds. The pipeline configuration files are in [.azp/](.azp/) which reference
-templates in [.azp/templates/](.azp/templates/).
+Visit the [workflows] page to find recent builds. The pipeline configuration
+files are in [.github/workflows/] which are built from templates in templates in
+[.github/workflows/templates/] using `make_workflows.py`.
```

Comment:
OK, will do. 

#982 configures `pre-commit` to run `make_workflows.py` automatically, so at least the two will always be consistent. Though it may lead to some developers losing modifications if they change the generated files. I did add comments to the files to help prevent this.